lib/core/test/unit/Cardano/Wallet/Primitive/FeeSpec.hs

```diff
@@ -567,13 +663,58 @@ genTxOut coins = do
     outs <- vector n
     return $ zipWith TxOut outs coins
 
-genSelection :: NonEmpty TxOut -> Gen CoinSelection
-genSelection outs = do
+genSelection :: Gen CoinSelection
+genSelection = do
+    outs <- choose (1, 10) >>= vector >>= genTxOut
+    genSelectionFor (NE.fromList outs)
+
+-- Like 'genSelection', but allows for having empty input and output.
+genSelection' :: Gen CoinSelection
+genSelection' = frequency
+    [ (4, genSelection)
+    , (1, do
+        reclaim_ <- genReclaim
+        pure $ mempty { reclaim = reclaim_ }
+      )
+    , (1, do
+        deposit_ <- genDeposit 100000
+        pure $ mempty { deposit = deposit_ }
+      )
+    ]
+
+genWithdrawal :: Gen Word64
+genWithdrawal = frequency
+    [ (3, pure 0)
+    , (1, oneof
```

Comment:
why this oneof?

Reply:
To favor some edge-cases a bit more in the generator. The type is a `Word64`, so something like `arbitrary` will span over the entire domain. Yet, we know that bugs are usually found at edges, so I've favored small values, close to the fee amount so see how it behaves when a bit smaller than the fee, or a bit bigger, and still, allowed for big values far from this edge.